webtester-core/src/test/java/integration/pageobjects/GenericSelectIntegrationTest.java

```diff
@@ -0,0 +1,131 @@
+package integration.pageobjects;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import integration.AbstractWebTesterIntegrationTest;
+
+import info.novatec.testit.webtester.api.annotations.IdentifyUsing;
+import info.novatec.testit.webtester.api.exceptions.WrongElementClassException;
+import info.novatec.testit.webtester.pageobjects.GenericSelect;
+import info.novatec.testit.webtester.pageobjects.PageObject;
+
+
+public class GenericSelectIntegrationTest extends AbstractWebTesterIntegrationTest {
+
+    GenericSelectTestPage page;
+
+    @Before
+    public void initPage() {
+        page = getBrowser().create(GenericSelectTestPage.class);
+    }
+
+    @Override
+    protected String getHTMLFilePath() {
+        return "html/pageobjects/genericSelect.html";
+    }
+
+    /* get all option texts */
+
+    @Test
+    public final void testThatListWithOptionTextsIsReturnedCorrectly_singleSelect() {
+        List<String> texts = page.singleSelect.getAllOptionTexts();
+        assertThat(texts.get(0).equals("one"), is(true));
+        assertThat(texts.get(1).equals("two"), is(true));
+        assertThat(texts.get(2).equals("three"), is(true));
+    }
+
+    @Test
+    public final void testThatListWithOptionTextsIsReturnedCorrectly_multiSelect() {
+        List<String> texts = page.multiSelect.getAllOptionTexts();
+        assertThat(texts.get(0).equals("one"), is(true));
+        assertThat(texts.get(1).equals("two"), is(true));
+        assertThat(texts.get(2).equals("three"), is(true));
+    }
+
+    @Test
+    public final void testThatListWithOptionTextsIsReturnedCorrectly_emptySelect() {
+        List<String> optionTexts = page.emptySelect.getAllOptionTexts();
+        assertThat(optionTexts.isEmpty(), is(true));
```

Comment:
`org.hamcrest.Matchers.empty`

All Hamcrest matchers have a factory method within the `Matchers` class.